File: codeforces/710D.cc
#include <iostream>
#include <cmath>

using namespace std;

typedef long long ll_t;

long long gcd_ext( ll_t a, ll_t b, ll_t& x, ll_t& y )
{
    ll_t t, g;
    if( b == 0 )
    {
        x = 1;
        y = 0;
        return a;
    }

    g = gcd_ext( b, a%b, x, y );
    t = x;
    x = y;
    y = t - a/b*y;

    return g;
}

int main()
{
    long long a1, b1, a2, b2, L, R;
    cin >> a1 >> b1 >> a2 >> b2 >> L >> R;

    ll_t A = a1;
    ll_t B = a2;
    ll_t C = abs(b2 - b1);

    ll_t X, Y;
    ll_t G = gcd_ext( A, B, X, Y );

    if( C % G )
    {
        cout << 0 << endl;
        return 0;
    }

    X = X*C/G;
    Y = Y*C/G;

    // X + B/G*k
    // Y - A/G*k
    // a1 * (X + B/G*k) + b1 >= L
    // a2 * (Y - A/G*k) + b2 >= L

    ll_t lo = ceil( (L - b1 - a1 * X) * G / double( B * a1 ) );
    ll_t hi = floor( (R - b1 - a1 * X) * G / double( B * a1 ) );

    ll_t ans = 0;

    // while( a2 * (Y - A/G*lo) + b2 < L ) lo++;

    while( a1 * (X + B/G*lo) + b1 <= R )
    {
        ll_t l = a2 * (Y - B/G*lo) + b2;
        ll_t r = a1 * (X + B/G*lo) + b1;
        cout << r << ' '<< l << endl;
        lo++;

        if( l >= L && l <= R )
            ans++;
    }

#ifdef LOCAL
    cout << G << ' ' << A << ' ' << B << ' ' << X << ' ' << Y << endl;
    cout << lo << ' ' << hi << endl;
#endif

    // cout << max( ll_t(0), hi - max( ll_t(0), lo ) + 1 ) << endl;

    cout << ans << endl;

    return 0;
}

File: codeforces/710D.py
import sys
from sys import stdin,stdout
import math

def gcd_ext(a, b):
    if b == 0:
        return (int(a),int(1),int(0))
    else:
        g,x,y = gcd_ext(b, a%b)
        t = x;
        x = y;
        y = t - a/b*y
        return int(g),int(x),int(y)

a1,b1,a2,b2,L,R = [ int(x) for x in stdin.readline().rstrip().split() ]

A = int(a1)
B = int(a2)
C = int(b2 - b1)

G,X,Y = gcd_ext( A, B )

if C % G != 0:
    print(0)
    sys.exit(0)

X = int(X*C/G)
Y = int(Y*C/G)

K = int(max( [ math.ceil( -X*G/float(B) ), math.ceil( Y*G/float(A) ) ] ))

# print("A",A,"B",B)
# print("G",G)
# print("X",X,"Y",Y)
# print("K",K)

# print("LOWER", (L - b1 - a1 * X) * G / float( B * a1 ), (L - b2 + a2 * Y) * G / float( A * a2 ))
# print("UPPER", (R - b1 - a1 * X) * G / float( B * a1 ), (R - b2 + a2 * Y) * G / float( A * a2 ))
# print("S", a1 * (X + B*K/G) + b1, "T", a2 * (-Y + A*K/G) + b2)

K = max( [ K, int(math.ceil( (L - b1 - a1 * X) * G / float( B * a1 ) )) ] )

U = math.floor( (R - b1 - a1 * X) * G / float( B * a1 ) + 0.00001 )

# print(K,U)

while a1 * (X + B*K/G) + b1 < L:
    K = K + 1

# print(K, a1 * (X + B*K/G) + b1)

if K > U or a1 * (X + B*K/G) + b1 > R:
    ANS = 0;
else:
    ANS = U - K + 1

print(max( [ 0, int(ANS) ] ))
